src/Storages/MergeTree/checkDataPart.cpp

```diff
@@ -399,18 +418,41 @@ IMergeTreeDataPart::Checksums checkDataPart(
 
         ReadSettings read_settings;
         read_settings.enable_filesystem_cache = false;
+        read_settings.enable_filesystem_cache_log = false;
+        read_settings.enable_filesystem_read_prefetches_log = false;
+        read_settings.page_cache = nullptr;
+        read_settings.load_marks_asynchronously = false;
+        read_settings.remote_fs_prefetch = false;
+        read_settings.page_cache_inject_eviction = false;
+        read_settings.use_page_cache_for_disks_without_file_cache = false;
```

Comment:
Also remote_fs_read_method -> read, local_fs_read_method -> pread? Because remote_fs_prefetch=false does not guarantee that threadpool is not used.